### PR TITLE
fix/rate oracle coingecko prices request

### PR DESCRIPTION
**Pull request by aarmoa**
**Before submitting this PR, please make sure**:

- [x] Your code builds clean without any errors or warnings
- [x] You are using approved title ("feat/", "fix/", "docs/", "refactor/")

**A description of the changes proposed in the pull request**:
Change in the RateOracle logic to get rates from Coingecko. Requests are now done using the `category` filter to ensure the client gets the values for the important tokens, ignoring tokens that are not used like NFTs.
The categories included in the requests are:
- cryptocurrency
- decentralized-exchange
- decentralized-finance-defi
- smart-contract-platform
- stablecoins
- wrapped-tokens

**Tests performed by the developer**:
Change unit tests to reflect the changes in the logic.
Manually tested using the `rate` command, querying for individual token prices and pair prices.

**Test summary**:
- Install feature branch without any issue
- Generate certs no issue
- Create a gateway v1 and start gateway no issue
- Connect ethereum with no issue
- Balance able to detect without any issue
- Config rate_oracle_source coingecko
- Current prices are shown when using rate tokens or pair with wrapped tokens
- Create amm_arb with weth-dai with no issue

**Tips for QA testing**:
After configuring Coingecko as the rate source (`config rate_oracle_source coingecko`) run amm arbitrage using wrapped tokens.
Check the client can provide prices using the `rate` command.

Fixes https://github.com/hummingbot/hummingbot/issues/3555

This PR comes from https://github.com/CoinAlpha/hummingbot/pull/159

PRP: https://snapshot.org/#/hbot-prp.eth/proposal/0x2fb919c8df0856b24f8d18f39d5e0ec911cb7ecb03a37daf70fcccf22c245766

### CODE REVIEW

**Review by rapcmia — APPROVED** [review]
Test performed:
- Install feature branch, ok 
- Set rate_oracle_source to coingecko
- Check rates --pair(-p) and --token(-t) of wrapped tokens (WETH, WBTC, WMATIC, etc..), all ok 
- Setup simple AMM-arbitrage using kovan testnet
- Setup simple XEMM strategy using binance and gateio using WBTC pairs
- Enable/disable use_oracle_conversion_rate and started the bot, no errors found 
- Tested using Gateway-v1 on both source and docker build, all ok

**Comment by aarmoa** [comment]
@JeremyKono the PRP for this PR closed on April 6th and QA has approved it. Please merge it into development, we need the change available in that branch for Gateway V2

**Comment by JeremyKono** [comment]
@aarmoa we are still waiting for this PR to be merged into `development` this is the last PR for the April release: https://github.com/hummingbot/hummingbot/pull/5181. After that, I will merge in all PRs that will go in to May release.

**Pull request merged**
